velox/connectors/Connector.h

```diff
@@ -250,7 +279,8 @@ class Connector {
   virtual std::shared_ptr<DataSink> createDataSink(
       RowTypePtr inputType,
       std::shared_ptr<ConnectorInsertTableHandle> connectorInsertTableHandle,
-      ConnectorQueryCtx* connectorQueryCtx) = 0;
+      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx,
+      std::optional<std::shared_ptr<WriteProtocol>> writeProtocol) = 0;
```

Comment:
std::shared_ptr already can be null, hence, no need to wrap it in std::optional. Doing so requires handling additional states, e.g. std::optional set, but the value is null.

Reply:
Got you